File: sdk-python/examples/greedy.py
"""Greedy bot — always moves toward the nearest cheese."""

import random

from pyrat_sdk import Bot, Context, Direction, GameState


class Greedy(Bot):
    name = "Greedy"
    author = "PyRat SDK"

    def think(self, state: GameState, ctx: Context) -> Direction:
        candidates = state.nearest_cheeses()
        if candidates:
            result = random.choice(candidates)
            if result.directions:
                return result.directions[0]
        return Direction.STAY


if __name__ == "__main__":
    Greedy().run()

File: sdk-rust/examples/greedy.rs
use pyrat_sdk::{Bot, Context, Direction, GameState, Options};

struct Greedy;

impl Options for Greedy {}

impl Bot for Greedy {
    fn think(&mut self, state: &GameState, _ctx: &Context) -> Direction {
        match state.nearest_cheese(None) {
            Some(result) if !result.path.is_empty() => result.path[0],
            _ => Direction::Stay,
        }
    }
}

fn main() {
    pyrat_sdk::run(Greedy, "Greedy", "PyRat");
}
